backend/src/main/java/com/bakdata/conquery/models/query/queryplan/aggregators/specific/EventDateUnionAggregator.java

```diff
@@ -19,18 +22,26 @@
  * provided date restriction.
  *
  */
-@RequiredArgsConstructor
+@Data
 @ToString(of = {"requiredTables"})
 public class EventDateUnionAggregator extends Aggregator<CDateSet> {
 
-	private final Set<Table> requiredTables;
+	@NonNull
+	private Set<Table> requiredTables = Collections.emptySet();
+	private QPNode owner;
+	private final CDateSet set = CDateSet.createEmpty();
+
 	private ValidityDate validityDateColumn;
-	private CDateSet set = CDateSet.createEmpty();
 	private CDateSet dateRestriction;
 
 	@Override
 	public void collectRequiredTables(Set<Table> requiredTables) {
-		requiredTables.addAll(this.requiredTables);
+		requiredTables.addAll(getRequiredTables());
```

Comment:
Das würde doch reichen oder?
```suggestion
		if(owner != null) {
		{
				requiredTables.addAll(owner.getRequiredTables());
		}
```